articles/tutorials/advanced/2d_shaders/02_hot_reload/index.md

```diff
@@ -0,0 +1,280 @@
+
+[!code-xml[](./snippets/snippet-2-14.xml?highlight=2)]
+
+To experiment with the system, re-run the following command:
+
+[!code-sh[](./snippets/snippet-2-15.sh)]
+
+And then cause some sort of compiler-error in the `grayscaleEffect.fx` file, such as adding the line, `"tunafish"` to the top of the file. When you save it, you should see the terminal spit out an error containing information about the compilation failure, 
+
+```
+ error X3000: unrecognized identifier 'tunafish'
+```
+
+Remove the `"tunafish"` line and save again, and the watch program should log some lines similar to these, 
+```
+  dotnet watch : Started
+    C:/proj/MonoGame.Samples/Tutorials/2dShaders/src/02-Hot-Reload-System/DungeonSlime/Content/effects/grayscaleEffect.fx
+    Copying Content...
+  
+  Build succeeded.
+      0 Warning(s)
+      0 Error(s)
+
+  Time Elapsed 00:00:01.40
+  dotnet watch : Exited
+  dotnet watch : Waiting for a file to change before restarting dotnet...
+
+```
+
+
+## Reload shaders in-game
+
+Now anytime the `.fx` files are modified, they will be recompiled and copied into the game's runtime folder. However, the game itself does not know to _reload_ the `Effect` instances. In this section, we will create a utility over the [`ContentManager`](xref:Microsoft.Xna.Framework.Content.ContentManager) to respond to these dynamic file updates. 
+
+It is important to make a distinction between assets the game _expects_ to be reloaded and assets that the game does _not_ care about reloading. This tutorial will demonstrate how to create an explicit system where individual assets opt _into_ being _hot reloadable_, rather than creating a system where all assets automatically handle dynamic reloading. 
+
+### Extending Content Manager
+
+Currently, the `grayscaleEffect.fx` is being loaded in the `GameScene` 's `LoadContent()` method like this:
+
+[!code-csharp[](./snippets/snippet-2-16.cs)]
+
+The `.Load()` function in the existing `ContentManager` is almost sufficient for our needs, but it returns a regular `Effect`, which has no understanding of the dynamic nature of the new content workflow. 
+
+Create a new _Content_ folder within the _MonoGameLibrary_ project, add a new file named `ContentManagerExtensions.cs`, and add the following code for the foundation of the new system:
+
+[!code-csharp[](./snippets/snippet-2-17.cs)]
+
+Now, you can create an [extension method](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) for the existing `MonoGame`'s `ContentManager` class:
+
+[!code-csharp[](./snippets/snippet-2-18.cs)]
```

Comment:
Several of the using statements in this class are unused, can er keep it clean please.

Reply:
I will fix